### ISSUE TITLE
TLEN: minor deviation from SAM specification

### ISSUE THREAD

**Issue by jts** [nanoporetech/dorado#109]
This is about as pedantic as an issue can get but writing the read length to the SAM TLEN field is not correct. The relevant section of the specification says:

```
[...]
It is set as 0 for a single-segment template or when the information is unavailable (e.g., when the first or last segment of a multi-segment template is unmapped or when the two are mapped to different reference sequences).
[...]
```

**Comment by vellamike** [comment]
You're right, that is pedantic!! Reading through the spec, TLEN seems to be defined in terms of alignment.

I think a pragmatic solution which respects the spec but keeps this information would be to set TLEN to zero and include the read length as a separate tag. What do you think?

**Comment by jts** [comment]
I think it would be fine as a tag (although it can be calculated from SEQ/cigar) so isn't strictly necessary imo

**Comment by cjw85** [comment]
This becomes important as soon as we talk about things like Pore-C (or any other multi-segment) reads. It should be changed in line with the specification. I doubt anyone is (or should be) using the field as it stands; I was in unaware it was being set!

**Comment by iiSeymour** [comment]
Great point @cjw85 - fixed in d49b73a5e6b2cffaa8382dfcb11721ac023c3f90
